rust/src/nasl/builtin/network/socket.rs

```diff
@@ -534,81 +397,123 @@ impl NaslSockets {
+                            .ok()
+                        } else {
+                            None
+                        };
+                        if let Ok(tcp) = TcpConnection::connect(
+                            addr,
+                            port,
+                            tls,
+                            timeout,
+                            bufsz,
+                            get_retry(context),
+                        ) {
                             // TODO: Set port transport
+                            tcp
+                        } else {
+                            continue;
                         }
                     }
-                }
-                // IP
-                Some(OpenvasEncaps::Ip) => {
-                    if let Ok(fd) = self.open_sock_tcp_ip(context, addr, port, bufsz, timeout, None)
-                    {
-                        fds.push(self.add(fd))
-                        // TODO: Set port transport
-                    }
-                }
-                // Unsupported transport layer
-                None | Some(OpenvasEncaps::Max) => {
-                    return Err(FunctionErrorKind::WrongArgument(format!(
-                        "unsupported transport layer: {transport}(unknown)"
-                    )))
-                }
-                // TLS/SSL
-                Some(tls_version) => match tls_version {
-                    OpenvasEncaps::Tls12 | OpenvasEncaps::Tls13 => {
-                        let fd =
-                            self.open_sock_tcp_tls(context, addr, port, bufsz, timeout, vhost)?;
-                        fds.push(self.add(fd))
+                    // IP
+                    Some(OpenvasEncaps::Ip) => {
+                        if let Ok(tcp) = TcpConnection::connect(
+                            addr,
+                            port,
+                            None,
+                            timeout,
+                            bufsz,
+                            get_retry(context),
+                        ) {
+                            // TODO: Set port transport
+                            tcp
+                        } else {
+                            continue;
+                        }
                     }
-                    _ => {
+                    // Unsupported transport layer
+                    None | Some(OpenvasEncaps::Max) => {
                         return Err(FunctionErrorKind::WrongArgument(format!(
-                            "unsupported transport layer: {transport}{tls_version}"
+                            "unsupported transport layer: {transport}(unknown)"
                         )))
                     }
-                },
-            }
+                    // TLS/SSL
+                    Some(tls_version) => match tls_version {
+                        OpenvasEncaps::Tls12 | OpenvasEncaps::Tls13 => {
+                            if let Ok(tls_conf) = Self::get_tls_conf(context) {
```

Comment:
nit: can you separate those blocks and may give them a name? This is hard to read and the else branch is always continue.

Reply:
I agree. Might be a case for `.and_then`